demo/handler.h

```diff
@@ -229,6 +240,17 @@ class Handler : public content_analysis::sdk::AgentEventHandler {
     std::cout << "  Filepath: " << file_path << std::endl;
     std::cout << "  Machine user: " << machine_user << std::endl;
     std::cout << "  Email: " << email << std::endl;
+    if (request.has_print_data() && !print_data_file_path_.empty()) {
+      int res = WriteHandleContentToFile((void*)request.print_data().handle(),
```

Comment:
As is, this requires the agent to be platform specific.  That is `WriteHandleContentToFile` (which part of the demo agent) is platform specific, which means the "agent" is platform specific.  So far this SDK has hidden all platform specific code behind interfaces.

What do you think of the following:

1/ Create an interface like:
```
class Client {
  ...
  class ScopedPrintData {
   public:
    static std::unique_ptr<ScopedPrintData> Create(const ContentAnalysisRequest& request);

    virtual ~ScopedPrintData() = default;
    virtual const char* data() = 0;
    virtual size_t size() = 0;
  }
  ...
}
```
2/ Have three impls, `ScopedPrintDataMac`, `ScopedPrintDataPosix`, `ScopedPrintDataWin`. In the Windows case, the ctor calls `MapViewOfFile` and gets the pointer to the data.  The dtor calls `CloseHandle`.

3/ Then demo code at lines 243-253 could be written something like:
```
  if (request.has_print_data() && !print_data_file_path_.empty()) {
    auto print_data = ScopedPrintData::Create(request);
    std::ofstream file(print_data_file_path_);
    file.write(print_data->data(), print_data->size());
  }
```

Reply:
Sorry Dom, to correct an error in my comment above: Instead of nesting inside `Client`, I meant to say nest inside of `ContentAnalysisEvent`.

Also, it might be possible to completely hide `ScopedPrintData` from the agent code.  For example, we could expose a method like `const char* ContentAnalysisEvent::GetPrintData()` that returns a pointer to the data, mapping it into the address only as needed using something like `ScopedPrintData`.  And when the event goes out of scope the memory can be unmapped and the handle closed and such.  This would also make it easier to mock out and write tests.  Thoughts?

Reply:
I am not sure this makes the code simpler, it might if other platforms work the same as windows but it may be that Mac or Linux ends up with a different workflow that's not as suitable to using Scoped* classes.

I think something better is to tweak the signature of the new function to something like:
`int WritePrintDataToFile(PrintData data, const std::string& path)`

That will make it easy to write reusable code since each platform implementation of the function can use different sub-fields of the proto instead of having odd (void*) casts everywhere, and building these functions with generic signatures like that will make it easier to implement and switch to Scoped* classes later on if we decide to. Thoughts?

Reply:
I've added the handle-closing code in `ContentAnalysisEventWin`, this way it should be closed within a sane lifetime. Let me know what you think of the current overall implementation.

Reply:
This is good, but the dtor of `ScopedPrintHandleWin` also closes.  I think you forgot to remove the latter?